src/common/Compromised.jsx

```diff
@@ -0,0 +1,31 @@
+import axios from "axios";
+import { generateSHA1Hash } from "../lib/encryption.ts";
+
+const PWNED_API_URL = "https://api.pwnedpasswords.com/range/";
+
+async function checkIsPasswordCompromised(password) {
+  const hash = await generateSHA1Hash(password);
+  const shortHash = hash.substring(0, 5);
+  const hashSuffix = hash.substring(5).toUpperCase();
+
+  try {
+    const response = await axios.get(`${PWNED_API_URL}${shortHash}`);
+
+    if(response.data) {
+      const results = response.data.split('\r\n');
+
+      for(const result of results) {
+        const [hashResult, numCompromises] = result.split(':');
+        if(hashResult === hashSuffix) {
+          return numCompromises;
+        }
+      }
+    }
+    return "0";
```

Comment:
It's unlikely that the response code would be 200 without any result.

The way the API works, you send only the first 5 characters of a SHA-1 hash of your password (so you don't expose a crackable version of it on the network).

The API returns a list of pwned passwords whose hashes have the 5-character prefix. Since there are so many passwords in the database, there's almost always some matches. However, I thought I should handle an empty result list, which should result in a return value "0".